08-read-write-plot.qmd

```diff
@@ -332,7 +335,162 @@ gpd.read_file('output/world_many_layers.gpkg', layer='world2').head(1)
 
 ### Raster data {#sec-data-output-raster}
 
-...
+To write a raster file using `rasterio`, we need to pass a raster file path to `rasterio.open`, in writing (`'w'`) mode. This implies creatign a new empty file (or overwriting an existing one). As opposed to read (`'r'`, the default) mode, the `rasterio.open` function needs quite a lot of information, in addition to the file path and mode:
+
+* An array with the raster values
+* Metadata describing the raster format and spatial properties
+
+The metadata needs to specify the following properties:
+
+* `driver`---The file format (The recommendation is `'GTiff'` for GeoTIFF)
+* `height`---Number of rows
+* `width`---Number of columns
+* `count`---Number of bands
+* `nodata`---The value which represents “No Data”, if any
+* `dtype`---The raster data type, one of numpy types (e.g., `np.int64`)
+* `crs`---The CRS, using an EPSG code (e.g., `4326`)
```

Comment:
Is it possible to use other codes than EPSG?

Reply:
Yes:
https://py.geocompx.org/07-reproj.html#sec-coordinate-reference-systems